DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.WebAPI/Controllers/ActivityController.cs

```diff
@@ -0,0 +1,50 @@
+using System.Linq;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Activity.External.WebAPI.Extensions;
+using GitHubExtension.Activity.External.WebAPI.Models;
+using GitHubExtension.Activity.External.WebAPI.Queries;
+using Microsoft.AspNet.Identity;
+
+namespace GitHubExtension.Activity.External.WebAPI.Controllers
+{
+    [Authorize]
+    public class ActivityController : ApiController
+    {
+        const string YouHaveNoRepositorySelected = "You have no repository selected";
+        const string CurrentProjectNameClaimType = "CurrentProjectName";
+        const string AccessTokenClaimType = "ExternalAccessToken";
+
+        public ActivityController(IGitHubEventsQuery eventsQuery)
+        {
+            _eventsQuery = eventsQuery;
+        }
+
+        private readonly IGitHubEventsQuery _eventsQuery;
+
+        [Route(ExternalActivityRoutes.GetGitHubActivityRoute)]
+        public async Task<IHttpActionResult> GetGitHubActivity([FromUri] int page)
+        {
+            var claimsIdentity = User.Identity as ClaimsIdentity;
+
+            string fullRepositoryName = claimsIdentity.FindFirstValue(CurrentProjectNameClaimType);
+            if (fullRepositoryName == null)
+                return BadRequest(YouHaveNoRepositorySelected);
+            
+            Claim tokenClaim = claimsIdentity.FindFirst(AccessTokenClaimType);
+            
+            IEnumerable<GitHubEventModel> events =
+                await _eventsQuery.GetGitHubEventsAsync(fullRepositoryName, tokenClaim.Value, page);
+            List<GitHubEventModel> eventModels = events.ToList();
+
+            if (eventModels.Count == 0)
+                return NotFound();
+            int? numberOfPages = _eventsQuery.GetNumberOfPages();
```

Comment:
Sure, from my point of view you can do two requests on Git Hub API to get this information. This two requests will similar BUT you can extract different type of info. Clarified?